File: python/semantic_kernel/exceptions/agent_exceptions.py
# Copyright (c) Microsoft. All rights reserved.


from semantic_kernel.exceptions.kernel_exceptions import KernelException


class AgentException(KernelException):
    """Base class for all agent exceptions."""

    pass


class AgentFileNotFoundException(AgentException):
    """The requested file was not found."""

    pass


class AgentInitializationException(AgentException):
    """An error occurred while initializing the agent."""

    pass


class AgentExecutionException(AgentException):
    """An error occurred while executing the agent."""

    pass


class AgentInvokeException(AgentException):
    """An error occurred while invoking the agent."""

    pass


class AgentChatException(AgentException):
    """An error occurred while invoking the agent chat."""

    pass


class AgentChatHistoryReducerException(AgentException):
    """An error occurred while reducing the chat history."""

    pass


class AgentThreadInitializationException(AgentException):
    """An error occurred while initializing the agent thread."""

    pass


class AgentThreadOperationException(AgentException):
    """An error occurred while performing an operation on the agent thread."""

    pass

File: nodejs/semantic-kernel/exceptions/agent-exceptions.ts
/**
 * Base class for all agent exceptions.
 */
export class AgentException extends Error {
  constructor(message: string) {
    super(message)
    this.name = 'AgentException'
  }
}

/**
 * The requested file was not found.
 */
export class AgentFileNotFoundException extends AgentException {
  constructor(message: string) {
    super(message)
    this.name = 'AgentFileNotFoundException'
  }
}

/**
 * An error occurred while initializing the agent.
 */
export class AgentInitializationException extends AgentException {
  constructor(message: string) {
    super(message)
    this.name = 'AgentInitializationException'
  }
}

/**
 * An error occurred while executing the agent.
 */
export class AgentExecutionException extends AgentException {
  constructor(message: string) {
    super(message)
    this.name = 'AgentExecutionException'
  }
}

/**
 * An error occurred while invoking the agent.
 */
export class AgentInvokeException extends AgentException {
  constructor(message: string) {
    super(message)
    this.name = 'AgentInvokeException'
  }
}

/**
 * An error occurred while invoking the agent chat.
 */
export class AgentChatException extends AgentException {
  constructor(message: string) {
    super(message)
    this.name = 'AgentChatException'
  }
}

/**
 * An error occurred while reducing the chat history.
 */
export class AgentChatHistoryReducerException extends AgentException {
  constructor(message: string) {
    super(message)
    this.name = 'AgentChatHistoryReducerException'
  }
}

/**
 * An error occurred while initializing the agent thread.
 */
export class AgentThreadInitializationException extends AgentException {
  constructor(message: string) {
    super(message)
    this.name = 'AgentThreadInitializationException'
  }
}

/**
 * An error occurred while performing an operation on the agent thread.
 */
export class AgentThreadOperationException extends AgentException {
  constructor(message: string) {
    super(message)
    this.name = 'AgentThreadOperationException'
  }
}
